bundles/binding/org.openhab.binding.maxcube/src/main/java/org/openhab/binding/maxcube/internal/message/MessageProcessor.java

```diff
@@ -0,0 +1,216 @@
+		}
+
+		MessageType messageType = getMessageType(line);
+
+		if (messageType == null) {
+			throw new UnsupportedMessageTypeException();
+		}
+
+		if ((this.currentMessageType != null) && (!messageType.equals(this.currentMessageType))) {
+			throw new IncompleteMessageException();
+		}
+
+		Boolean result = true;
+
+		switch (messageType) {
+		case H:
+			this.currentMessage = new H_Message(line);
+			break;
+		case C:
+			this.currentMessage = new C_Message(line);
+			break;
+		case L:
+			this.currentMessage = new L_Message(line);
+			break;
+		case S:
+			this.currentMessage = new S_Message(line);
+			break;
+		case M:
+			result = handle_M_MessageLine(line);
+			break;
+		default:
+		}
+
+		return result;
+	}
+
+	private Boolean handle_M_MessageLine(String line) throws UnprocessableMessageException, IncompleteMessageException, IncorrectMultilineIndexException {
+		Boolean result = false;
+
+		String[] tokens = line.split(Message.DELIMETER);  //M:00,01,xyz.....
+
+		try {
+			Integer index = Integer.valueOf(tokens[0].replaceFirst("M:", ""));  //M:00
+			Integer counter = Integer.valueOf(tokens[1]);  //01
+
+			if (this.numberOfRequiredLines == null) {
+				switch (counter) {
+				case 0:
+					throw new UnprocessableMessageException();
+				case 1:
+					this.currentMessage = new M_Message(line);
+					result = true;
+					break;
+				default:
+					this.numberOfRequiredLines = counter;
+					this.currentMessageType = MessageType.M;
+					if (index == 0) {
+						this.receivedLines.add(line);
+					} else {
+						throw new IncorrectMultilineIndexException();
+					}
+				}
+			} else {
+				if ((!counter.equals(this.numberOfRequiredLines)) || (!(index == this.receivedLines.size()))) {
+					throw new IncorrectMultilineIndexException();
+				}
+				
+				receivedLines.add(tokens[2]);
+				
+				if (index+1 == receivedLines.size()) {
+					String newLine = "";
+					for (String curLine : receivedLines)
+						newLine += curLine;
+					this.currentMessage = new M_Message(newLine);
+					result = true;
+				}
+			}
+		} catch (IncorrectMultilineIndexException ex) {
+			throw ex;
+		} catch (Exception ex) {
+			throw new UnprocessableMessageException();
+		}
+
+		return result;
+	}
+
+	/**
+	 * Pulls the message from the stack when there is one available. This needs
+	 * to be done before next line can be added into message processor. When
+	 * message is pulled, the message processor is reseted and ready to process
+	 * next line.
+	 * 
+	 * @return Message
+	 * @throws NoMessageAvailableException
+	 *             when there was no message on the stack
+	 */
+	public Message pull() throws NoMessageAvailableException {
+		Message result = null;
+
+		if (this.currentMessage == null) {
+			throw new NoMessageAvailableException();
+		} else {
+			result = this.currentMessage;
+			reset();
+		}
+
+		return result;
+	}
+
+	/**
+	 * Processes the raw TCP data read from the MAX protocol, returning the
+	 * corresponding Message.

```

Comment:
what I mean to say is that the description does not match what is actually returned
```
* Processes the raw TCP data read from the MAX protocol, returning the 
* corresponding Message 
```
but it is returning a MessageType instead of a Message